crates/rome_js_analyze/src/analyzers/nursery/no_prototype_builtins.rs

```diff
@@ -0,0 +1,150 @@
+use rome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_js_syntax::{AnyJsExpression, JsCallExpression, JsCallExpressionFields, TextRange};
+use rome_rowan::AstNodeList;
+
+declare_rule! {
+    /// Disallow direct use of `Object.prototype` builtins.
+    ///
+    /// ECMAScript 5.1 added `Object.create` which allows creation of object with a custom prototype.
+    /// This pattern is often used for objects used as Maps. However this pattern can lead to errors
+    /// if something else relies on prototype properties/methods.
+    /// Moreover, the methods could be shadowed, this can lead to random bugs and denial of service
+    /// vulnerabilities. For example, calling `hasOwnProperty` directly on parsed json like `{"hasOwnProperty": 1}` could lead to vulnerabilities.
+    /// To avoid subtle bugs like this, you should call these methods from `Object.prototype`.
+    /// For example, `foo.isPrototypeof(bar)` should be replaced with `Object.prototype.isPrototypeof.call(foo, "bar")`
+    /// As for the `hasOwnProperty` method, `foo.hasOwnProperty("bar")` should be replaced with `Object.hasOwn(foo, "bar")`.
+    /// `Object.hasOwn()` is native replacement for `Object.prototype.hasOwnProperty()`. Refer to [the Object.hasOwn() documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn).
```

Comment:
```suggestion
    /// ECMAScript 5.1 added `Object.create` which allows the creation of an object with a custom prototype.
    /// This pattern is often used for objects used as Maps. However, this pattern can lead to errors
    /// if something else relies on prototype properties/methods.
    /// Moreover, the methods could be shadowed, this can lead to random bugs and denial of service
    /// vulnerabilities. For example, calling `hasOwnProperty` directly on parsed JSON like `{"hasOwnProperty": 1}` could lead to vulnerabilities.
    /// To avoid subtle bugs like this, you should call these methods from `Object.prototype`.
    /// For example, `foo.isPrototypeof(bar)` should be replaced with `Object.prototype.isPrototypeof.call(foo, "bar")`
    /// As for the `hasOwn` method, `foo.hasOwn("bar")` should be replaced with `Object.hasOwn(foo, "bar")`.
```

I think it's better to just directly provide the example with `hasOwn`